extensions/amp-fx-flying-carpet/0.1/amp-fx-flying-carpet.js

```diff
@@ -110,15 +110,15 @@ class AmpFlyingCarpet extends AMP.BaseElement {
     const docHeight = viewport.getScrollHeight();
     // Hmm, can the page height change and affect us?
     user().assert(
-      layoutBox.top >= viewportHeight,
+      layoutBox.top >= viewportHeight * 0.8,
```

Comment:
/cc @jasti, we discussed this [before](https://github.com/ampproject/amphtml/issues/3416#issuecomment-224349480) and decided to collapse when flying carpet when the effect would be broken.